src/Forcings/advective_forcing.jl

```diff
@@ -0,0 +1,84 @@
+using Oceananigans.Advection: UpwindBiasedFifthOrder, div_Uc, div_𝐯u, div_𝐯v, div_𝐯w
+using Oceananigans.Fields: ZeroField, ConstantField
+using Adapt
+
+maybe_constant_field(u) = u
+maybe_constant_field(u::Number) = ConstantField(u)
```

Comment:
Defining the function twice?

Reply:
There are two methods: one "fallback" which just passes `u` through, and a second function that converts `u` to `ConstantField(u)` when `u` is an integer. This allows `u` to be indexed into in the advection scheme functions. (A `ConstantField` is a field that returns a constant when we write `u[i, j, k]`).

Reply:
I see, so basically the function just pass u if u is not a number (e.g. a Field). If u is a number, it will return a constant field.